File: M/670-maximum-swap/main.go
package main

import (
	"fmt"
	"sort"
)

func maximumSwap(num int) int {
	if num < 10 {
		return num
	}

	var buf []int
	for num > 0 {
		m := num % 10
		buf = append(buf, m)
		num /= 10
	}

	ss := make([]int, len(buf))
	copy(ss, buf)
	sort.Ints(ss)

	idx := -1
	for i := len(buf) - 1; i >= 0; i-- {
		if ss[i] != buf[i] {
			idx = i
			break
		}
	}
	if idx != -1 {
		idx2 := -1
		for i := 0; i < len(buf); i++ {
			if buf[i] == ss[idx] {
				idx2 = i
				break
			}
		}
		buf[idx], buf[idx2] = buf[idx2], buf[idx]
	}

	n := 0
	for i := len(buf) - 1; i >= 0; i-- {
		n = n*10 + buf[i]
	}
	return n
}

func main() {
	fmt.Println("a")
}

File: M/670-maximum-swap/main.rs
struct Solution {}

impl Solution {
  pub fn maximum_swap(num: i32) -> i32 {
    if num < 10 {
      return num;
    }

    let mut buf: Vec<i32> = Vec::new();
    let mut num2 = num;
    while num2 > 0 {
      buf.push(num2 % 10);
      num2 /= 10;
    }
    let mut sort = buf.clone();
    sort.sort();
    let mut idx: usize = buf.len() - 1;
    while idx > 0 {
      if buf[idx] != sort[idx] {
        break;
      }
      idx -= 1;
    }
    if idx == 0 {
      return num;
    }

    let mut idx2: usize = 0;
    while idx2 < buf.len() {
      if buf[idx2] == sort[idx] {
        break;
      }
      idx2 += 1;
    }
    let t = buf[idx2];
    buf[idx2] = buf[idx];
    buf[idx] = t;
    buf.reverse();
    let mut ret = 0;
    idx = 0;
    while idx < buf.len() {
      ret = ret * 10 + buf[idx];
      idx += 1;
    }
    ret
  }
}
